psycop/common/feature_generation/sequences/cohort_definer_to_prediction_times.py

```diff
@@ -34,14 +34,34 @@ def _polars_dataframe_to_patient_timestamp_mapping(
         dataframe: pl.DataFrame,
         id_col_name: str,
         patient_timestamp_col_name: str,
+        lookahead: dt.timedelta | None = None,
     ) -> dict[PATIENT_ID, list[dt.datetime]]:
+        """Maps a polars dataframe to a dictionary of {patient ids: timestamps}.
+
+        Args:
+            dataframe: Polars dataframe with patient ids and timestamps.
+            id_col_name: Name of the column with patient ids.
+            patient_timestamp_col_name: Name of the column with timestamps.
+            lookahead: If not None, only timestamps before the max timestamp minus the lookahead
+                will be included. E.g. for a lookahead of 2 years, this ensures we actually have 2
+                years of data to label the outcome.
+        """
+        if lookahead is not None:
+            max_timestamp: dt.datetime = dataframe[patient_timestamp_col_name].max()  # type: ignore
```

Comment:
Would be even better to get the max timestamp of (outcome_timestamps | prediction_time_timestamps | predictors), but this is much faster, and probably just as good.